crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -1282,6 +1286,85 @@ impl BeaconState {
         }
         Ok(())
     }
+
+    pub fn process_attestation(&mut self, attestation: &Attestation) -> anyhow::Result<()> {
+        ensure!(
+            attestation.data.target.epoch == self.get_previous_epoch()
+                || attestation.data.target.epoch == self.get_current_epoch(),
+            "Target epoch must be the previous or current epoch"
+        );
+
+        ensure!(
+            attestation.data.target.epoch == compute_epoch_at_slot(attestation.data.slot),
+            "Target epoch must match the computed epoch at slot"
+        );
+
+        ensure!(
+            attestation.data.slot + MIN_ATTESTATION_INCLUSION_DELAY <= self.slot,
+            "Attestation must be included after the minimum delay"
+        );
+
+        ensure!(
+            attestation.data.index
+                < self.get_committee_count_per_slot(attestation.data.target.epoch),
+            "Committee index must be within bounds"
+        );
+
+        let committee = self.get_beacon_committee(attestation.data.slot, attestation.data.index)?;
+        ensure!(
+            attestation.aggregation_bits.len() == committee.len(),
+            "Aggregation bits length must match committee size"
+        );
+
+        let participation_flag_indices = self.get_attestation_participation_flag_indices(
+            attestation.data.clone(),
+            self.slot - attestation.data.slot,
+        )?;
```

Comment:
can you make this function take a reference of attestation.data instead of cloning